bgpd/bgp_routemap.c

```diff
@@ -696,6 +696,58 @@ struct route_map_rule_cmd route_match_evpn_vni_cmd = {
 	"evpn vni", route_match_vni, route_match_vni_compile,
 	route_match_vni_free};
 
+/* `match evpn route-type' */
+
+/* Match function should return 1 if match is success else return
+   zero. */
+static route_map_result_t route_match_evpn_route_type(void *rule,
+						      struct prefix *prefix,
+						      route_map_object_t type,
+						      void *object)
+{
+	u_char route_type = 0;
+
+	if (type == RMAP_BGP) {
+		route_type = *((u_char *)rule);
+
+		if (route_type == prefix->u.prefix_evpn.route_type)
+			return RMAP_MATCH;
+	}
+
+	return RMAP_NOMATCH;
+}
+
+/* Route map `route-type' match statement. */
+static void *route_match_evpn_route_type_compile(const char *arg)
+{
+	u_char *route_type = NULL;
+
+	route_type = XMALLOC(MTYPE_ROUTE_MAP_COMPILED, sizeof(u_char));
+	if (!route_type)
```

Comment:
nit: `XMALLOC` calls `abort()` if it fails to allocate memory, so there's no need to check for errors here. Also, the `route_type` initialization to NULL is unnecessary.

Reply:
made the change.